src/UglyToad.PdfPig/Util/PatternParser.cs

```diff
@@ -20,7 +20,7 @@ public static PatternColor Create(IToken pattern, IPdfTokenScanner scanner, IRes
             if (DirectObjectFinder.TryGet(pattern, scanner, out StreamToken? fs))
             {
                 patternDictionary = fs.StreamDictionary;
-                patternStream = new StreamToken(fs.StreamDictionary, fs.Decode(filterProvider, scanner));
+                patternStream = new StreamToken(fs.StreamDictionary, fs.Decode(filterProvider, scanner).ToArray());
```

Comment:
I don't think you need thew `ToArray()¬, you can use the `ReadOnlyMemory` directly

Reply:
(same for all `fs.Decode(filterProvider, scanner)`)

Reply:
I'll double check these are the terminals of the decode / filter pipe (in which case, this will be safe) and will update.

Reply:
good catch. this is safe! done.